main.ts

```diff
@@ -286,22 +292,56 @@ export default class AutoLinkTitle extends Plugin {
     if (title.length < this.settings.maximumTitleLength + 3) {
       return title;
     }
-    const shortenedTitle = `${title.slice(0, this.settings.maximumTitleLength)}...`;
+    const shortenedTitle = `${title.slice(
+      0,
+      this.settings.maximumTitleLength
+    )}...`;
     return shortenedTitle;
+  };
+
+  public async fetchUrlTitleViaLinkPreview(url: string): Promise<string> {
+    try {

```

Comment:
Looks great!
If we add a check to ensure the `linkPreviewApiKey` is 32 characters long so we don't hit their API with invalid requests when the user hasn't put in an API key I'd be up for merging this!